cmake/SetupClangTidy.cmake

```diff
@@ -32,13 +32,14 @@
 #         public and protected member variables are fine
 # -readability-uppercase-literal-suffix: we're okay with lower case
 # -readability-const-return-type: complains about decltype(auto)
+# -readability-redundant-declaration: false positives on friends
 #
 # Notes:
 # misc-move-const-arg: we keep this check because even though this gives
 #                      a lot of annoying warnings about moving trivially
 #                      copyable types, it warns about moving const objects,
 #                      which can have severe performance impacts.
-set(CLANG_TIDY_IGNORE_CHECKS "*,-cppcoreguidelines-no-malloc,-llvm-header-guard,-google-runtime-int,-readability-else-after-return,-misc-noexcept-move-constructor,-misc-unconventional-assign-operator,-cppcoreguidelines-c-copy-assignment-signature,-modernize-raw-string-literal,-hicpp-*,-android-*,-cert-err58-cpp,-google-default-arguments,-fuchsia-*,-performance-noexcept-move-constructor,-modernize-use-trailing-return-type,-cert-oop54-cpp,-misc-definitions-in-headers,-cppcoreguidelines-macro-usage,-cppcoreguidelines-avoid-magic-numbers,-modernize-use-nodiscard,-readability-magic-numbers,-bugprone-exception-escape,-cert-msc32-c,-misc-non-private-member-variables-in-classes,-cppcoreguidelines-avoid-c-arrays,-cppcoreguidelines-non-private-member-variables-in-classes,-cert-msc51-cpp,-bugprone-macro-parentheses,-readability-uppercase-literal-suffix,-readability-const-return-type")
+set(CLANG_TIDY_IGNORE_CHECKS "*,-cppcoreguidelines-no-malloc,-llvm-header-guard,-google-runtime-int,-readability-else-after-return,-misc-noexcept-move-constructor,-misc-unconventional-assign-operator,-cppcoreguidelines-c-copy-assignment-signature,-modernize-raw-string-literal,-hicpp-*,-android-*,-cert-err58-cpp,-google-default-arguments,-fuchsia-*,-performance-noexcept-move-constructor,-modernize-use-trailing-return-type,-cert-oop54-cpp,-misc-definitions-in-headers,-cppcoreguidelines-macro-usage,-cppcoreguidelines-avoid-magic-numbers,-modernize-use-nodiscard,-readability-magic-numbers,-bugprone-exception-escape,-cert-msc32-c,-misc-non-private-member-variables-in-classes,-cppcoreguidelines-avoid-c-arrays,-cppcoreguidelines-non-private-member-variables-in-classes,-cert-msc51-cpp,-bugprone-macro-parentheses,-readability-uppercase-literal-suffix,-readability-const-return-type,-readability-redundant-declaration")
```

Comment:
Now I'm even more confused. We've been ignorning the `readability-redundant-declaration` with friends because of the bug you noticed, but always using `NOLINTNEXTLINE`. However, the check is useful in general. In this PR it started showing up again and you had changed to a blanket `NOLINT`. The thing I'm not clear on is did the `NOLINTNEXTLINE` stop work, the name of the check change, or another check that does the same thing get added?

Reply:
Sorry, as I tried to clarify in my last comment, the `NOLINTNEXTLINE` previously (mysteriously) worked despite a spelling mistake in the error name that I only noticed after @kidder 's comment. I added the this to the cmake following your comment on the same thread, having misunderstood the request, so I'll just remove it now.

The change here will then just be correcting a spelling mistake in the `NOLINTNEXTLINE`

Reply:
Ah okay, sorry for the confusion!